packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -34,7 +34,7 @@ abstract class ApplicationPackage {
   File get packagesFile => null;
 
   @override
-  String toString() => displayName;
+  String toString() => displayName ?? 'application';
```

Comment:
Why can this ever be null?

Reply:
Replaced `'application'` by `id` which is asserted not to be null. This avoids silent errors. Thanks.